Avoid matching artifacts of other groups

mackdk opened the pull request:
After the introduction of the `--parse-pom` flag, the way of handling the `groupId` of the artifacts has changed. This has introduced a bug because if the flag is omitted (i.e. when downloading dependencies for ant) the code currently look for existing jars in the whole designated repository folder, instead of only in the configured default group. This means `obs-to-maven` won't donwload a dependency if a jar already exists with a different group but same `mtime`. This scenario is currently happening with `commons-jexl.jar`.

This PR addresses this issue by using the previous search path when the flag is omitted. When the flag is specified, we cannot include the group id in the path to search artifacts because we don't know the group id until the dependency is downloaded and the group id is not the same anyway. Potentially, we could still hit the bug in `--parse-pom` mode if two packages contains an artifact with the same `artifactId`, different `groupId`, but the jar file has the same `mtime`.

cbosdo reviewed (changes requested):
Can you please set the description as commit message? This makes archeology much easier and provider-independent!

mackdk committed:
Avoid matching artifacts of other groups

After the introduction of the --parse-pom flag, the way of handling the
groupId of the artifacts has changed. This has introduced a bug because
if the flag is omitted (i.e. when downloading dependencies for ant)
the code currently look for existing jars in the whole designated
repository folder, instead of only in the configured default group.
This means obs-to-maven won't donwload a dependency if a jar already
exists with a different group but same mtime.

This commit addresses this issue by using the previous search path
when the flag is omitted.

When the flag is specified, we cannot include the group id in the path
to search artifacts because we don't know the group id until the
dependency is downloaded and the group id is not the same anyway.
Potentially, we could still hit the bug in parse-pom mode if two
packages contains an artifact with the same artifactId, different
groupId, but the jar file has the same mtime.
